Tools/machines/karolina-it4i/karolina_gpu_warpx.profile.example

```diff
@@ -0,0 +1,66 @@
+# please set your project account
+export proj=""  # change me!
+
+# remembers the location of this script
+export MY_PROFILE=$(cd $(dirname $BASH_SOURCE) && pwd)"/"$(basename $BASH_SOURCE)
+if [ -z ${proj-} ]; then echo "WARNING: The 'proj' variable is not yet set in your $MY_PROFILE file! Please edit its line 2 to continue!"; return; fi
+
+# required dependencies
+module load GCCcore/11.3.0
+module load CUDA/11.7.0
+module load CMake/3.23.1-GCCcore-11.3.0
+module load OpenMPI/4.1.4-GCC-11.3.0-CUDA-11.7.0
+
+# optional: for QED support with detailed tables
+module load Boost/1.79.0-GCC-11.3.0
+
+# optional: for openPMD and PSATD+RZ support
+module load OpenBLAS/0.3.20-GCC-11.3.0
+export CMAKE_PREFIX_PATH=${HOME}/sw/karolina/gpu/hdf5-1.14.1.2:$CMAKE_PREFIX_PATH
+export CMAKE_PREFIX_PATH=${HOME}/sw/karolina/gpu/c-blosc-1.21.1:$CMAKE_PREFIX_PATH
+export CMAKE_PREFIX_PATH=${HOME}/sw/karolina/gpu/adios2-2.8.3:$CMAKE_PREFIX_PATH
+export CMAKE_PREFIX_PATH=${HOME}/sw/karolina/gpu/blaspp-master:$CMAKE_PREFIX_PATH
+export CMAKE_PREFIX_PATH=${HOME}/sw/karolina/gpu/lapackpp-master:$CMAKE_PREFIX_PATH
+
+export LD_LIBRARY_PATH=${HOME}/sw/karolina/gpu/hdf5-1.14.1.2/lib64:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=${HOME}/sw/karolina/gpu/c-blosc-1.21.1/lib64:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=${HOME}/sw/karolina/gpu/adios2-2.8.3/lib64:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=${HOME}/sw/karolina/gpu/blaspp-master/lib64:$LD_LIBRARY_PATH
+export LD_LIBRARY_PATH=${HOME}/sw/karolina/gpu/lapackpp-master/lib64:$LD_LIBRARY_PATH
+
+# optional: CCache (not found)
+#module load ccache
+
+# optional: for Python bindings or libEnsemble
+module load Python/3.10.4-GCCcore-11.3.0-bare
+
+if [ -d "${HOME}/sw/karolina/gpu/venvs/warpx-gpu" ]
+then
+  source ${HOME}/sw/karolina/gpu/venvs/warpx-gpu/bin/activate
+fi
+
+# an alias to request an interactive batch node for one hour (TODO)
+#   for parallel execution, start on the batch node: srun <command>
+#alias getNode="salloc -N 1 --ntasks-per-node=4 -t 1:00:00 -q interactive -C gpu --gpu-bind=single:1 -c 32 -G 4 -A $proj"
+# an alias to run a command on a batch node for up to 30min
+#   usage: runNode <command>
+#alias runNode="srun -N 1 --ntasks-per-node=4 -t 0:30:00 -q interactive -C gpu --gpu-bind=single:1 -c 32 -G 4 -A $proj"
+
+# PBS is not well configured to properly address GPU visibility
+# per MPI rank. Thus, we define the node layouts here.
+export GPUS_PER_SOCKET=2
```

Comment:
Karolina's GPU nodes have 8 A100 each, see [compute-nodes](https://docs.it4i.cz/karolina/compute-nodes/), so shouldn't this read:

```bash
export GPUS_PER_SOCKET=4
export GPUS_PER_NODE=8
```
?

Reply:
Yes, but we will probably use visibility settings for this if we find an equivalent to SLURM_LOCALID in PBS

Reply:
Does this [page](https://docs.it4i.cz/general/slurmtopbs/) from their docs help?

Reply:
The most similar concept in `PBS` is probably the use of the `PBS_TASKNUM` environment variable, which provides the ID of the task in a parallel job, but there are some key differences. `PBS_TASKNUM` starts from 1 (not 0 like `SLURM_LOCALID`), and also, it refers to the global rank across all nodes, not the local rank on the individual node.

Reply:
We can use
MPI_LOCALRANKID
According to https://docs.it4i.cz/general/slurmtopbs/